src/deparse.c

```diff
@@ -0,0 +1,227 @@
+            // Q:: do I need to worry about collation owner, namespace etc?
+            appendStringInfo(qs, " COLLATE %s", quote_identifier(NameStr(pg_collation_row->collname)));
+        }
+
+        if (attr.attnotnull)
+            appendStringInfoString(qs, " NOT NULL");
+
+        if (attr.atthasdef) {
+            // Q::: shall I optimize this?
+            for(int i = 0; i < rel_descr->constr->num_defval;i++) {
+                AttrDefault attr_def = rel_descr->constr->defval[i];
+                if (attr_def.adnum == attr.attnum) {
+                    /* These macros allow the collation argument to be omitted (with a default of
+                    * InvalidOid, ie, no collation).  They exist mostly for backwards
+                    * compatibility of source code.
+                    ^^ Q:: looks like DirectFunctionCall is for backward compatibility, shall I still use it?*/
+                    // Q:: shall I keep separate variable? shall I define it in the beginning of the function?
+                    char *attr_default = TextDatumGetCString(DirectFunctionCall2(pg_get_expr,
+                                                            CStringGetTextDatum(attr_def.adbin),
+                                                            ObjectIdGetDatum(reloid)));
+                    appendStringInfo(qs, " DEFAULT %s",attr_default); // Q:: shall this be quoted? if so how?
+                    // quoting it as literal makes things like this 'val'::text -> ''val'::text ''::text
+                    break;
+                }
+            }
+        }
+    }
+
+    relation_close(pg_collation, AccessShareLock);
+    relation_close(pg_type, AccessShareLock);
+    appendStringInfoString(qs, "\n)");
+    appendStringInfoString(qs, ";\n");
+}
+
+// adapted from chunk_constraint.c:chunk_constraints_add_inheritable_constraints
+/* Add constraint informaiton of the given Relation to the given StringInfo
+ * as valid SQL ALTER TABLE statements.
+*/
+void ct_deparse_constraints(StringInfo qs, Relation table_rel) {
+    Oid reloid = table_rel->rd_id;
+	ScanKeyData skey;
+	Relation	rel;
+	SysScanDesc scan;
+	HeapTuple	htup;
+    Oid constroid;
+    char * constrdef;
+	ScanKeyInit(&skey,
+				Anum_pg_constraint_conrelid,
+				BTEqualStrategyNumber, F_OIDEQ, reloid);
+
+	rel = heap_open(ConstraintRelationId, AccessShareLock);
+	scan = systable_beginscan(rel, ConstraintRelidIndexId, true,
+							  NULL, 1, &skey);
+
+	while (HeapTupleIsValid(htup = systable_getnext(scan)))
+	{
+		Form_pg_constraint pg_constraint = (Form_pg_constraint) GETSTRUCT(htup);
+        if (pg_constraint->condeferred ||pg_constraint->condeferrable)
+            ereport(ERROR,
+                (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                    errmsg("Table with OID %u has deferred or deferrable constraints."
+                           " These are not supported in deparsing",
+                            reloid)));
+
+
```

Comment:
You should be able to call HeapTupleGetOid(htup) to get the OID so no need to call  get_relation_constraint_oid